### PR TITLE
Add support for PHP 8.4

### PR DESCRIPTION
**Pull request by Atul-glo35265**
|    Q          |   A
|-------------- | ------
| Documentation | yes/no
| Bugfix        | yes/no
| BC Break      | yes/no
| New Feature   | yes/no
| RFC           | yes/no
| QA            | yes/no

### CODE REVIEW

**Comment by Atul-glo35265** [comment]
@froschdesign Kindly review the PR once and do needful.

**Comment by Atul-glo35265** [comment]
@gsteel, kindly review this PR once and merge it, if possible.

**Review by alexmerlin — APPROVED** [review]
Looks good to me! 👍

File: src/AutoDiscover.php [hunk 1]
```diff
@@ -625,11 +625,19 @@ public function toXml()
...
 
     /**
-     * Handle WSDL document.
+     * Handle WSDL document
+     *
+     * @param  string|null|callable $responseHandler
+     * @return void
      */
-    public function handle()
+    public function handle($responseHandler = null)
```
Comment (froschdesign) [inline src/AutoDiscover.php]: @alexmerlin 
The class is not final therefore this cannot be correct.

Comment (Atul-glo35265) [reply src/AutoDiscover.php]: @froschdesign We have made the class final. Please review once.

Comment (samsonasik) [reply src/AutoDiscover.php]: @Atul-glo35265 make class final is bc break, if it required, uses `func_get_args()` may can handle that, see https://3v4l.org/8na14

Comment (froschdesign) [reply src/AutoDiscover.php]: @Atul-glo35265 
Changing the method signature and making a class final represent a break in backwards compatibility.

The question that arises here is why was the method changed at all? Only for tests?

Comment (Atul-glo35265) [reply src/AutoDiscover.php]: @froschdesign There were 3 errors in [QA Checks (PHPUnit [8.4, lowest] failure](https://github.com/laminas/laminas-soap/actions/runs/11327465531/job/31498597871). We made changes in these respective methods to fix the issue.

I'll try to implement if we can achieve the same result by implementing func_get_args() as suggested by [samsonasik](https://github.com/samsonasik)

Comment (froschdesign) [reply src/AutoDiscover.php]: @Atul-glo35265 
> We made changes in these respective methods to fix the issue.

The error is not in the `AutoDiscover` class and `func_get_args` will not help here, as the logic of the method does not require an external parameter.
Check and correct the tests and not the implementation.

Comment (froschdesign) [reply src/AutoDiscover.php]: @Atul-glo35265 
Please use a newer version of PHPUnit, otherwise the problem persists.

Comment (Atul-glo35265) [reply src/AutoDiscover.php]: Yes @froschdesign. I'm trying the same on the local machine.

Comment (froschdesign) [reply src/AutoDiscover.php]: Change the version in the Composer configuration file. See:

- https://github.com/sebastianbergmann/phpunit/releases/tag/9.6.21
- https://github.com/sebastianbergmann/phpunit/releases/tag/9.6.18

Comment (Atul-glo35265) [reply src/AutoDiscover.php]: @froschdesign I've updated the version to 9.6.21 but the issue is still there.

Actually, the annotation **@runinseparateprocess** is causing the issue and throwing the exception. If we remove this annotation, issues related to _Header already sent_ start coming.

I tried to upgrade PHPUnit to version **^10.0.0**, but it seems we need to upgrade other dependencies as well.

Comment (froschdesign) [reply src/AutoDiscover.php]: @Atul-glo35265 
> I've updated the version to 9.6.21 but the issue is still there.

The version of PHPUnit must be changed in the `composer.json` file!

Comment (Atul-glo35265) [reply src/AutoDiscover.php]: @froschdesign I have updated it in composer.json file. Please check and suggest if any other changes are required.

Comment (froschdesign) [reply src/AutoDiscover.php]: Bump also all other packages in `require-dev` section.

https://getcomposer.org/doc/03-cli.md#bump

Comment (alexmerlin) [reply src/AutoDiscover.php]: If `bump`ing does not work, you can upgrade them manually. I just tested, they can be upgraded.

File: src/AutoDiscover.php [hunk 2]
```diff
@@ -18,6 +17,7 @@
 use function header;
 use function htmlspecialchars;
 use function is_array;
+use function is_callable;
```
Comment (alexmerlin) [inline src/AutoDiscover.php]:
```suggestion
```

**Review by alexmerlin — CHANGES_REQUESTED** [review]

File: src/AutoDiscover.php (hunk 2 above)
Comment (Atul-glo35265) [reply src/AutoDiscover.php]: Done

**Comment by froschdesign** [comment]
@Atul-glo35265 
The problem is `amphp/amp`, so instead of upgrading PHPUnit to version 10, which introduces new problems, add older versions of `amphp/amp` as a conflict:

```json
"conflict": {
    "amphp/amp":"<2.6.4"
}
```

**Comment by Atul-glo35265** [comment]
> @Atul-glo35265 The problem is `amphp/amp`, so instead of upgrading PHPUnit to version 10, which introduces new problems, add older versions of `amphp/amp` as a conflict:
> 
> ```json
> "conflict": {
>     "amphp/amp":"<2.6.4"
> }
> ```

Thank you @froschdesign. Please review once and do the needful.

**Comment by nishant04412** [comment]
@gsteel Please check and advise if this looks good or anything else needs to be addressed ?

**Comment by froschdesign** [comment]
@nishant04412 
Please stop pinging people. It is done when it is done.

---
You can have a look at the tests and maybe you can explain why the change is necessary.

**Comment by froschdesign** [comment]
@Atul-glo35265 
Please check the history to see if the test included this check from the beginning or if it was changed at some point.

**Comment by Atul-glo35265** [comment]
> @Atul-glo35265 Please check the history to see if the test included this check from the beginning or if it was changed at some point.

It was added by @glo71317 while adding support for PHP8.3. Please check [here](https://github.com/laminas/laminas-soap/commit/d710a9d2ed49472ca6abc769a79043c1e4f45d6d#diff-1c0b695e0d9e491a9331b9b7d278f9e1ae86ee4187b32fc146e1fe647e11da79L593).
It was causing test failure while adding support for PHP 8.4, so I reverted it.

**Review by alexmerlin — APPROVED** [review]

**Pull request merged**